api/API_VERSIONING.md

```diff
@@ -84,52 +100,43 @@ from clients (Envoy, gRPC) and a large number of control planes, ranging from si
-  human editable artifact.
-
-An example of how this might play out is that at the end of December in 2020, if a v4 major version
-is justified, we might freeze
-`envoy.config.bootstrap.v4alpha` and this package would then become the current stable major version
-`envoy.config.bootstrap.v4`. The `envoy.config.bootstrap.v3` package will become the previous stable
-major version and support for `envoy.config.bootstrap.v2` will be dropped from the Envoy
-implementation. Note that some transitively referenced package, e.g.
-`envoy.config.filter.network.foo.v2` may remain at version 2 during this release, if no changes were
-made to the referenced package. If no major version is justified at this point, the decision to cut
-v4 might occur at some point in 2021 or beyond, however v2 support will still be removed at the end
-of 2020.
-
-The implication of this API lifecycle and clock is that any deprecated feature in the Envoy API will
-retain implementation support for at least 1-2 years.
-
-We are currently working on a strategy to introduce minor versions
-(https://github.com/envoyproxy/envoy/issues/8416). This will bump the xDS API minor version on every
-deprecation and field introduction/modification. This will provide an opportunity for the control
-plane to condition on client and major/minor API version support. Currently under discussion, but
-not finalized will be the sunsetting of Envoy client support for deprecated features after a year
-of support within a major version. Please post to https://github.com/envoyproxy/envoy/issues/8416
-any thoughts around this.
+A new API minor version is expected to be released once a year, around Q1. This
+allows for gradual introduction of new capabilities and features that will be supported
+by clients, and by control planes.
+Following the release of a new minor version, the API lifecycle follows a deprecation clock.
+Envoy will support the most recent 3 minor versions, including the current. This
+entails that we provide at least 2 years of supported deprecated features to
+unset.
+
+The patch version will be incremented upon changes to the API proto files, such
+as adding a new enum value, or setting a field as deprecated. We expect frequent
+updates to the patch version.
+
+The following is a an example of the API versioning lifecycle. Assume that the
+current API version is v3.7.18. Assume that a new change (pull-request) updates the API as follows:
+`field1` is annotated as deprecated and `field2`, a new field, is added.
+As a consequence of this, the patch version is increased and the API version becomes v3.7.19.
+Upon sending a request for xDS resources, the client provides the resource type
+which includes its full package path which entails the API major version (v3), and
+the range of minor versions that the client supports (5 to 7) using the resource
+context parameters: `xds.api.oldest_minor_ver=5` and `xds.api.latest_minor_ver=7`.
```

Comment:
As above, best to have both Node and context parameter examples as many will use Node.

Reply:
Done.